src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -264,10 +270,70 @@ private int LineBreaksAfter(SyntaxToken currentToken, SyntaxToken nextToken)
             return 0;
         }
 
+        private static bool All<TNode>(SyntaxList<TNode> items, Func<TNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            bool result = true;
+            for (int i = 0; i < items.Count; i++)
+            {
+                result &= predicate(items[i]);
+            }
+
+            return result;
+        }
+
+        private static bool IsAutoAccessorList(SyntaxToken token)
+            => token.Parent is AccessorListSyntax accessorList &&
+                All(accessorList.Accessors, a => a.Body == null);
+
+        private static bool IsAutoAccessorList(SyntaxNode? node)
```

Comment:
>IsAutoAccessorList [](start = 28, length = 18)

Similar comment about the name, the function doesn't give an answer about the node that is given as an input.

Reply:
Actually I renamed both as "HasParentAutoAccessors", but I'll wait for your comments before committing

Reply:
> Actually I renamed both as "HasParentAutoAccessors", but I'll wait for your comments before committing

I suggest making the following changes:
- Instead of having two helpers have one helper that takes a SyntaxNode
- It shouldn't check the parent of the node, it should check the node itself
- The name of the method is "IsAccessorListWithoutAccessorsWithBlockBody" or something similar
- Call-sites should be adjusted to pass the right node (i.e. the parent) accordingly.

---
In reply to: [531003112](https://github.com/dotnet/roslyn/pull/49495#discussion_r531003112) [](ancestors = 531003112)